### DISCUSSION TITLE
More official supports on mobile platform please.

### DISCUSSION THREAD

**Discussion by wangdong20** [godotengine/godot-proposals#4318] [General Discussions]
I just use Godot to develop mobile game on Google play and App Store. Finally I made it, but I feel like Godot has very few support on mobile platform. Here is the list I feel can be added in Godot for mobile platform.
1. No official support for mobile ads platform like Admob, etc. Ad support is very important for mobile app developer, we want official supports for ads such as unity.
2. Output file size for android and iOS is too large. I built a small simple game with some pictures and sounds, but the export size is way beyond my expectation. In android is about 20+Mb and 30+Mb for iOS. I know it may have ways to optimize the output file but seems not big help. Can Godot provide a default option to optimize output file size over load speed or reverse in export setting? I tried other game engine like Libgdx to build game also, the output file is amazing small and running good. I think the output file size is more sensitive for mobile app users than other platform. I hope more improvements on optimization for output file size on mobile platform.

**Comment by mrjustaguy** [comment]
I don't know how it works on mobile, but on desktop, aside from what you make, you get 40-50 MBs in the engine executable that's copied, and to get less you have to compile out things you don't use to shrink the size of the executable, and this is probably also the case on mobile

**Comment by theraot** [comment]
In general I agree.

However, I want to say I don't think Admob or any other platform integration should be in core, instead they should be addons so you can choose which you want (I would rather Godot didn't pick favorites, in particular not among third party privative platforms). By the way, Mobile is not the only size sensible platform, web is too, and people making web games will have a different set of platform integrations.

Godot 4 has grown in size compared to Godot 3.x, but:

- With GDExtension there is idea of moving some parts to officially supported addons (and thus reducing the core size). See https://github.com/godotengine/godot-proposals/issues/3286

- Also there are proposals to streamlining the process of making a game, then figuring out what features it needs, then making a custom godot build for it, then building the game. See:

  - https://github.com/godotengine/godot-proposals/issues/1001
  - https://github.com/godotengine/godot-proposals/issues/726

*Plus any effort to reduce bloat, such the one Juan Linietsky announced on twitter regarding the rendering system: https://twitter.com/reduzio/status/1509451600166797318*

So, there is hope for the build size to go down.
